### PR TITLE
fix dataset tests & module name

### PR DESCRIPTION
**Pull request by joelberkeley**
main change: rename test_datasets.py to test_data.py, which I missed in an earlier PR where I renamed datasets.py to data.py

### CODE REVIEW

File: tests/unit/test_data.py [hunk 1]
```diff
@@ -73,13 +73,13 @@ def test_dataset_raises_on_initialisation_for_invalid_ranks(
 
 
 def test_dataset_getters() -> None:
-    query_points, observations = tf.zeros((3, 3)), tf.zeros((3, 3))
+    query_points, observations = tf.zeros((3, 2)), tf.ones((3, 1))
```
Comment (joelberkeley) [inline tests/unit/test_data.py:76]: this wasn't a very good test, without being able to differentiate between query points and observations, or axes

File: tests/unit/test_data.py [hunk 2]
```diff
@@ -73,13 +73,13 @@ def test_dataset_raises_on_initialisation_for_invalid_ranks(
...
     dataset = Dataset(query_points, observations)
     assert tf.reduce_all(dataset.query_points == query_points)
     assert tf.reduce_all(dataset.observations == observations)
 
 
-def test_concatenate_datasets() -> None:
+def test_dataset_concatenation() -> None:
```
Comment (joelberkeley) [inline tests/unit/test_data.py:82]: rename to align with other test names

File: tests/unit/test_data.py [hunk 3]
```diff
@@ -73,13 +73,13 @@ def test_dataset_raises_on_initialisation_for_invalid_ranks(
 
 
 def test_dataset_getters() -> None:
-    query_points, observations = tf.zeros((3, 3)), tf.zeros((3, 3))
+    query_points, observations = tf.constant([[0.0]]), tf.constant([[1.0]])
```
Comment (joelberkeley-secondmind) [inline tests/unit/test_data.py:76]: this wasn't a very good test, without being able to differentiate between query points and observations, or axes

Comment (joelberkeley) [inline tests/unit/test_data.py:76]: this wasn't a very good test, as there wasn't any difference between the example query points and observations

**Review by ChrisMorter — APPROVED** [review]
Looks good

**Pull request merged**